compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs

```diff
@@ -59,7 +59,7 @@ fn push_debuginfo_type_name<'tcx>(
     match *t.kind() {
         ty::Bool => output.push_str("bool"),
         ty::Char => output.push_str("char"),
-        ty::Str => output.push_str("str"),
+        ty::Str => output.push_str("str$"),
```

Comment:
Looks like this needs to be guarded by a `if cpp_like_debuginfo` check.

Reply:
Indeed.